frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -528,47 +573,17 @@ function ProblemDisplay(props: ProblemDisplayParams) {
 
           <LowMarginMediumText>Problem Inputs</LowMarginMediumText>
           {newProblem.problemInputs.map((input, index) => (
-            <InputTypeContainer key={generateRandomId()}>
-              <TextInput
-                value={newProblem.problemInputs[index].name}
-                onChange={(e) => handleInputChange(index,
-                  e.target.value, newProblem.problemInputs[index].type)}
-                disabled={!problemEditable}
-              />
-
-              <InlineErrorIcon
-                show={!validIdentifier(newProblem.problemInputs[index].name)}
-                onMouseEnter={() => setHoverVisible(true)}
-                onMouseMove={mouseMoveHandler}
-                onMouseLeave={() => setHoverVisible(false)}
-              >
-                error_outline
-              </InlineErrorIcon>
-
-              <PrimarySelect
-                onChange={(e) => handleInputChange(
-                  index,
-                  newProblem.problemInputs[index].name,
-                  ProblemIOType[e.target.value as keyof typeof ProblemIOType],
-                )}
-                value={problemIOTypeToString(newProblem.problemInputs[index].type)}
-                disabled={!problemEditable}
-              >
-                {
-                  Object.keys(ProblemIOType).map((key) => (
-                    <option key={key} value={key}>{key}</option>
-                  ))
-                }
-              </PrimarySelect>
-
-              {problemEditable ? (
-                <CancelTextButton
-                  onClick={() => deleteProblemInput(index)}
-                >
-                  ✕
-                </CancelTextButton>
-              ) : null}
-            </InputTypeContainer>
+            <ProblemInputDisplay
+              // eslint-disable-next-line react/no-array-index-key
+              key={index}
+              input={input}
```

Comment:
Was refactoring this code into ProblemInputDisplay part of the solution to the bug fix you mentioned? If not I feel like the amount of props needed to pass this component makes the code feel a bit rigid - maybe instead we could either figure out some way to reduce the number of props, or even instead move the entirety of the options sidebar into its own component? 

For example if we move the options sidebar into its own component I think it'd only require 4 props (problemEditable, editMode, newProblem, setNewProblem - tho in the near future I'm thinking we can move newProblem to the Redux store which would cut this to 2) while also cutting the size of the ProblemDisplay file almost in half

Reply:
The refactoring was not part of the solution - and that's a fair point. I might lean towards just undoing the creation of this new component right now, since you suggested we may make even more changes to this whole page. I'm thinking we could do the refactoring / separating components there as necessary, but let me know if I should do it here instead.

Reply:
Yeah I think that sounds good - we can do that once we get around to that refactoring PR

Reply:
Ok, I reverted this change but kept the small fix that resolves #157. 👍